File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/InstaDeath.lua
function GetName()
  return "InstaDeath"
end

function GetType()
  return Debuff
end

function GetHookPoints()
  return { }
end

function OnAttachment(char)
  char:TakeTrueDamage(char:GetCurrentHP() * 3)
  this:RemoveFrom(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/InstaDeath.cs
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class InstaDeath : Affliction
    {
        public override string Name { get { return "InstaDeath"; } }

        public override AfflictionType Type { get { return AfflictionType.Debuff; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new HookPoint[]
                {
                };
            }
        }

        protected override void OnAttachment(ICharacter character)
        {
            character.TakeTrueDamage(character.CurrentHP * 10);
            RemoveFrom(character);
        }
    }
}